File: tests/fixtures/fixture_ts/src/lib/payment.ts
import { Order, calculateTotal } from '@models/order';

export function processPayment(order: Order): { success: boolean; transactionId: string } {
  const total = calculateTotal([order.total]);
  return { success: total > 0, transactionId: crypto.randomUUID() };
}

export function refundPayment(transactionId: string): boolean {
  return transactionId.length > 0;
}

File: tests/fixtures/fixture_rust/src/services/payment.rs
use crate::models::order::Order;

pub fn process_payment(order: &Order) -> Result<bool, String> {
    if order.total() <= 0.0 {
        return Err("Invalid amount".to_string());
    }
    Ok(true)
}

pub fn refund_payment(order: &Order) -> Result<bool, String> {
    if order.total() <= 0.0 {
        return Err("Cannot refund zero amount".to_string());
    }
    Ok(true)
}
